mm2src/kdf_walletconnect/src/lib.rs

```diff
@@ -358,10 +359,11 @@ impl WalletConnectCtxImpl {
             .await?;
 
         let (tx, rx) = oneshot::channel();
+        // insert request to map with a reasonable expiration time of 30 minutes
         self.pending_requests
             .lock()
             .expect("pending request lock shouldn't fail!")
```

Comment:
nit: we can just unwrap here. no need for justification for mutex unwraps.

Reply:
WalletConnect doesn't limit timeout it all depends on the timeout we want to set for each request. 
I agree 30minsis a very long expiration time.